### PR TITLE
feat: show list of suppliers while creating purchase order from material request

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1199536704671393)

### CODE REVIEW

File: erpnext/stock/doctype/material_request/material_request.py [hunk 1]
```diff
@@ -393,13 +394,37 @@ def get_default_supplier_query(doctype, txt, searchfield, start, page_len, filte
...
 	for d in doc.items:
 		item_list.append(d.item_code)
-
 	return frappe.db.sql("""select default_supplier
 		from `tabItem Default`
 		where parent in ({0}) and
 		default_supplier IS NOT NULL
 		""".format(', '.join(['%s']*len(item_list))),tuple(item_list))
 
+# @frappe.whitelist()
```
Comment (hrwX) [inline erpnext/stock/doctype/material_request/material_request.py]:
```suggestion
@frappe.whitelist()
```

File: erpnext/stock/doctype/material_request/material_request.py [hunk 2]
```diff
@@ -393,13 +394,37 @@ def get_default_supplier_query(doctype, txt, searchfield, start, page_len, filte
...
+def get_suppliers(doctype, txt, searchfield, start, page_len, filters):
+	doc = frappe.get_doc("Material Request", filters.get("doc"))
+	item_list = []
+	for d in doc.items:
+		item_list.append(d.item_code)
+	data = frappe.db.sql("""select supplier
+		from `tabItem Supplier`
+		where parent in ({0})
+		""".format(', '.join(['%s']*len(item_list))),tuple(item_list))
```
Comment (hrwX) [inline erpnext/stock/doctype/material_request/material_request.py]:
```suggestion
	data = frappe.db.sql("""select supplier
		from `tabItem Supplier`
		where parent in ({0})
		""".format(', '.join(['%s']*len(item_list))),tuple(item_list))
```
Use ORM

File: erpnext/stock/doctype/material_request/material_request.py [hunk 3]
```diff
@@ -393,13 +394,37 @@ def get_default_supplier_query(doctype, txt, searchfield, start, page_len, filte
...
+def get_rate(doc, supplier):
+	doc = frappe.get_doc("Material Request", doc)
+	item_list = []
+	for d in doc.items:
+		item_list.append(d.item_code)
+	data = frappe.db.sql("""select item_supplier.price_list_rate, item_supplier.uom
+		from `tabItem Supplier` as item_supplier
+		where parent in %(item)s and
+		item_supplier.supplier = %(supplier)s
+		""".format(', '.join(['%s']*len(item_list))), {'item': tuple(item_list), 'supplier': supplier})
```
Comment (hrwX) [inline erpnext/stock/doctype/material_request/material_request.py]: Use ORM

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/material_request/material_request.py [hunk 4]
```diff
@@ -393,13 +394,37 @@ def get_default_supplier_query(doctype, txt, searchfield, start, page_len, filte
...
+		item_list.append(d.item_code)
+	data = frappe.db.sql("""select supplier
+		from `tabItem Supplier`
+		where parent in ({0})
+		""".format(', '.join(['%s']*len(item_list))),tuple(item_list))
+	return data
+
+@frappe.whitelist()
+def get_rate(doc, supplier):
+	doc = frappe.get_doc("Material Request", doc)
```
Comment (rvpasari) [inline erpnext/stock/doctype/material_request/material_request.py:424]: Docstring/comments

**Review by rvpasari — CHANGES_REQUESTED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**